sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,311 @@
+          'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+          'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath',
+          'sqeuclidean', 'yule']
+
+        See the documentation for scipy.spatial.distance for details on these
+        metrics.
+
+    metric_params : dict, optional (default = None)
+        Additional keyword arguments for the metric function.
+
+    contamination : float in (0., 0.5), optional (default=0.1)
+        The amount of contamination of the data set, i.e. the proportion
+        of outliers in the data set. When fitting it is used to define the
+        threshold on the decision function.
+
+    n_jobs : int, optional (default = 1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only :meth:`k_neighbors` and :meth:`kneighbors_graph` methods.
+    """
+    def __init__(self, n_neighbors=5, algorithm='auto', leaf_size=30,
+                 metric='minkowski', p=2, metric_params=None,
+                 contamination=0.1, n_jobs=1):
+        self._init_params(n_neighbors=n_neighbors,
+                          algorithm=algorithm,
+                          leaf_size=leaf_size, metric=metric, p=p,
+                          metric_params=metric_params, n_jobs=n_jobs)
+
+        self.get_n_jobs_ = _get_n_jobs(self.n_jobs)
+        self.contamination = contamination
+
+    def fit(self, X, y=None):
+        """Fit the model using X as training data
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix, BallTree, KDTree}
+            Training data. If array or matrix, shape [n_samples, n_features],
+            or [n_samples, n_samples] if metric='precomputed'.
+        """
+        super(LocalOutlierFactor, self).fit(X)
+
+        # Compute decision_function over training samples to define threshold_:
+        self.decision_function_train_ = self.decision_function()
+
+        self.threshold_ = -sp.stats.scoreatpercentile(
+            -self.decision_function_train_, 100. * (1. - self.contamination))
+        return self
+
+    def predict(self, X=None):
+        """Predict Local Outlier Factor of X.
+
+        The local outlier factor (LOF) of a sample captures its
+        supposed `degree of abnormality'.
+        It is the average of the ratio of the local reachability density of
+        a sample and those of its k-nearest neighbors.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features), default=None
+            The query sample or samples to compute the Local Outlier Factor
+            wrt to the training samples. If None, makes prediction on the
+            training data without considering them as their own neighbors.
+
+        Returns
+        -------
+        is_inlier : array of shape (n_samples,)
+            Returns 1 for anomalies/outliers and -1 for inliers.
+        """
```

Comment:
Before doing `predict` you have to be sure that you have done `fit` maybe it could be interesting to add a function that check the value of `self.decision_function_train_` and `self.threashold_`.